src/date/expand-pattern.js

```diff
@@ -57,16 +147,20 @@ return function( options, cldr ) {
 				} else {
 					type = "short";
 				}
-				result = combineDateTime( type,
-					cldr.main([
-						"dates/calendars/gregorian/dateTimeFormats/availableFormats",
-						dateSkeleton
-					]),
-					cldr.main([
-						"dates/calendars/gregorian/dateTimeFormats/availableFormats",
-						timeSkeleton
-					])
+				dateSkeleton = getBestMatchPattern(
+					"dates/calendars/gregorian/dateTimeFormats/availableFormats",
+					dateSkeleton
 				);
+				timeSkeleton = getBestMatchPattern(
+					"dates/calendars/gregorian/dateTimeFormats/availableFormats",
+					timeSkeleton
+				);
```

Comment:
You might need use `getBestMatchPattern()` for each date and time before the `type` assignment above. For example, `MMMMh` should give `July, 8 PM` (type medium) instead of `July at 8 PM` (type long). Because, "MMMM" will be converted into "LLLL".